openedx/adg/lms/student/views.py

```diff
@@ -64,3 +73,120 @@ def get_social_oauth_providers_data(self):
                     })
 
         return context
+
+
+class DashboardView(LoginRequiredMixin, TemplateView):
+    """
+    View for dashboard page
+    """
+
+    template_name = 'student_account/dashboard.html'
+
+    def get_context_data(self, **kwargs):
+        """
+        Add the context regarding social auth providers to the view
+        """
+        context = super(DashboardView, self).get_context_data(**kwargs)
+
+        completed_courses_details, inprogress_courses_details = self.get_courses_overview_and_status()
+
+        webinars = self.get_webinars_overview()
+
+        certificates = self.get_certificates_overview()
+
+        context.update({
+            'user': self.request.user,
+            'completed_courses': completed_courses_details,
+            'inprogress_courses': inprogress_courses_details,
+            'webinars': webinars,
+            'certificates': certificates,
+        })
+        return context
+
+    def get_courses_overview_and_status(self):
+        """
+            Returns: list of completed and in progress course(s) with details
+        """
+        enrolled_course_ids = list(
+            CourseEnrollment.objects.filter(user=self.request.user).values_list('course', flat=True)
+        )
+        completed_courses = list(
+            PersistentCourseGrade.objects.filter(Q(user_id=self.request.user.id) and ~Q(passed_timestamp=None)).values(
+                'course_id', 'percent_grade')
+        )
+
+        completed_course_ids = [completed_courses[index]['course_id'] for index in range(len(completed_courses))]
```

Comment:
If this list is only created to find the course grade from the course id, then instead of two list simply create a dictionary to optimize better. As .index() method in loop for lookup, using dict will be faster & preferable.